### PR TITLE
Nonuc error

### PR DESCRIPTION
**Pull request by DevinSmithWork**
Throws a new userErrorHalt when: The depositor's primary group is non-uc and there's no grants linked to the pub

### CODE REVIEW

File: transform.rb [hunk 1]
```diff
@@ -462,10 +465,43 @@ def elementsToJSON(oldData, elemPubID, submitterEmail, metaHash, ark, feedFile)
...
 
+  # Check for non-uc depositors and non-uc items without grants.
+  # This is easier to handle after the tranforms.
+
   # All done.
   return data
 end
 
+###################################################################################################
+# If the depositor is non-uc but there's no non-uc grant, throw a user error halt
```
Comment (myucekul) [inline transform.rb:476]: The comment says that a non-UC depositor must have a non-UC grant. But it looks like the logic is looking for UC's RGPO grants.

File: transform.rb [hunk 2]
```diff
@@ -462,10 +465,43 @@ def elementsToJSON(oldData, elemPubID, submitterEmail, metaHash, ark, feedFile)
...
+  if depositorGroup == 'rgpo-nonuc'
+    
+    if funderTypeDisplayName == nil
+      userErrorHalt(ark, "Deposit not accepted: publication must be linked to a RGPO grant (CBCP, CBCRP, TRDRP, etc). Please return to the \"link grant\" screen or contact us for assistance.")
+    else
+      nonUCGrantFound = false
+
+      # Split the funder types; loop; set the boolean and break if found.
+      funderTypeDisplayName.split("|").each { |funder|
+        if ["RGPO (CBCRP) Award", "RGPO (CHRP) Award", "RGPO (TRDRP) Award", "RGPO (UCRI) Award"].include? funder
```
Comment (myucekul) [inline transform.rb:489]: Are these funding names expected to stay constant?

File: transform.rb [hunk 3]
```diff
@@ -462,10 +465,43 @@ def elementsToJSON(oldData, elemPubID, submitterEmail, metaHash, ark, feedFile)
   # Custom Citation Field
   metaHash.key?("custom-citation") and data[:customCitation] = metaHash.delete("custom-citation")
 
+  # Check for non-uc depositors and non-uc items without grants.
```
Comment (myucekul) [inline transform.rb:468]: Extra comment lines?

File: transform.rb (hunk 1 above)
Comment (DevinSmithWork) [reply transform.rb:476]: I worded this oddly -- fixing. Should be "If the depositor is non-uc, but there's not RGPO-related grant, throw an error"

File: transform.rb (hunk 2 above)
Comment (DevinSmithWork) [reply transform.rb:489]: I'm not sure -- I could make this more generalized by looking only for the RGPO program strings (CHRP, CBCRP, RGPO, etc)

**Pull request merged**